src/componentLogic/Router.ts

```diff
@@ -0,0 +1,22 @@
+import { html, computed, BaseRouter } from "../mod.ts";
+import type { Routes } from "./BaseRouter.ts"
+
+export class Router extends BaseRouter {
```

Comment:
So, there's some trickiness with a path-based router; the issue being that you can't intercept route changes due to href.

The solution I worked out for this was to capture click events on anchor tags and translate them to a router navigation.

Reply:
A second thing to consider: how might one go about navigating to a path from within JS?

Reply:
A third thing you'd want to look at is popstate events.

Reply:
So of course this will have to be handled at a backend level to support the various endpoints (see the `a.ts` file I added) which is why I’m using a http server to power this app, so that can handle those endpoints

Reply:
ah, I thought you were doing something like SPA

Reply:
A 'SPA' can only handle explicit routes (eg `/users`, `/users/new`) if it has the resource for it in the backend right?

Reply:
What do you mean? The frontend can edit the current path arbitrarily without firing a proper navigation. However, if it's not a hash router, the backend needs to be configured to return the same page for ever path, which lets the frontend take care of figuring out what should be displayed for a given route. Not sure what you mean by your question, though.

Reply:
Thats exactly what i mean actually - the whole idea of `Router` is that it doesn't use hrefs, every resource in the backend will render the same html file, which will call the same app.js file - only different is, the Router will determine which component to use based on the uri